UNRELEASED.md

```diff
@@ -1,7 +1,12 @@
 # UNRELEASED
 
+_If `@skyscanner/eslint-config-skyscanner` has a major version change then `@skyscanner/stylelint-config-skyscanner` should also have a major version change, as explained in [the docs](./docs/eslint-as-dependency.md)_
```

Comment:
I've added a note here to help us remember, as if I, who was there at the time we made this decision, can have completely forgotten, it is very easy for others to do the same - so with an inline reminder?